neqo-bin/src/server/mod.rs

```diff
@@ -580,6 +489,60 @@ enum Ready {
     Timeout,
 }
 
+pub async fn firefox() -> Res<()> {
+    let args: Vec<String> = env::args().collect();
+    if args.len() < 2 {
+        eprintln!("Wrong arguments.");
+        exit(1)
+    }
+
+    // Read data from stdin and terminate the server if EOF is detected, which
+    // means that runxpcshelltests.py ended without shutting down the server.
+    thread::spawn(|| loop {
+        let mut buffer = String::new();
+        match io::stdin().read_line(&mut buffer) {
+            Ok(n) => {
+                if n == 0 {
+                    exit(0);
+                }
+            }
+            Err(_) => {
+                exit(0);
+            }
+        }
+    });
+
+    init_db(PathBuf::from(args[1].clone())).unwrap();
+
+    let local = tokio::task::LocalSet::new();
+
+    local.spawn_local(ServersRunner::firefox(ServerType::Http3, 0)?.run());
+    local.spawn_local(ServersRunner::firefox(ServerType::Http3Fail, 0)?.run());
+    local.spawn_local(ServersRunner::firefox(ServerType::Http3Ech, 0)?.run());
+
+    let proxy_port = match env::var("MOZ_HTTP3_PROXY_PORT") {
+        Ok(val) => val.parse::<u16>().unwrap(),
+        _ => 0,
+    };
+    local.spawn_local(ServersRunner::firefox(ServerType::Http3Proxy, proxy_port)?.run());
+    local.spawn_local(ServersRunner::firefox(ServerType::Http3NoResponse, 0)?.run());
+
+    // TODO
+    // println!(
+    //     "HTTP3 server listening on ports {}, {}, {}, {} and {}. EchConfig is @{}@",
```

Comment:
FYI
These output lines are used here.
https://searchfox.org/mozilla-central/rev/e69f323af80c357d287fb6314745e75c62eab92a/testing/mozbase/mozserve/mozserve/servers.py#117

Reply:
Thank you @KershawChang. Didn't know this is being parsed. Your hint saved me quite a bit of debugging.